crates/aptos-crypto/src/ed25519/ed25519_keys.rs

```diff
@@ -65,6 +67,15 @@ impl Ed25519PrivateKey {
         let sig = expanded_secret_key.sign(message.as_ref(), &public_key.0);
         Ed25519Signature(sig)
     }
+
+    /// Derive the actual scalar represented by the secret key.
```

Comment:
We are breaking the safety abstraction of `aptos-crypto` by doing this.

Can you add a TODO here?

```
TODO: We are temporarily breaking the abstraction here and exposing the SK scalar. In the future, we should add traits for encryption inside aptos-crypto so that we can both sign and decrypt with an Ed25519PrivateKey
```

Reply:
fixed